snuba/utils/concurrent.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from concurrent.futures import Future
+from contextlib import contextmanager
+from threading import Lock, Thread
+from typing import Callable, Generic, Iterator, Optional, TypeVar
+
+
+T = TypeVar("T")
+
+
+def execute(
+    function: Callable[[], T], name: Optional[str] = None, daemon: Optional[bool] = None
+) -> Future[T]:
+    """
+    Execute a callable in a background thread, returning a ``Future`` object
+    that can be used to wait for the callable to exit (or check if it is
+    still running.)
+
+    This is similar to using ``concurrent.futures.ThreadPoolExecutor.submit``
+    but does not require a thread pool. Since each invocation will create
+    (and eventually shut down) new thread, this function is better suited for
+    long-lived background tasks (and generally IO bound ones, at that) than
+    distributing large workloads.
+    """
+    future: Future[T] = Future()
+
+    def run() -> None:
+        if not future.set_running_or_notify_cancel():
+            return
+
+        try:
+            result = function()
```

Comment:
I'm not totally sure I understand the question — it'll be whatever the return value of `function` is defined as per `Callable[[], T]`. If the function isn't typed, this would become `Callable[[], Any]` (translated into `Future[Any]`.) If the function doesn't have a return value (like specifically no `return` statement), it'd be `Callable[[], None]` (again translated into `Future[None]`) as long as the function itself is defined with `-> None`. (This one in particular isn't any different than any other return value defined with `->`. `Any` is the only real special case here.)

Reply:
_ If the function doesn't have a return value (like specifically no return statement), it'd be Callable[[], None]_

That answered my question.